platform/darwin/src/http_file_source.mm

```diff
@@ -88,29 +83,29 @@ void cancel() {
 public:
     Impl() {
         @autoreleasepool {
-            NSURLSessionConfiguration *sessionConfig = [MGLNetworkConfiguration sharedManager].sessionConfiguration;
+            NSURLSessionConfiguration *sessionConfig = MGLNativeAppleInterfaceTransmitter.shared.sessionConfiguration;
```

Comment:
Longer-term, I would prefer to avoid having & going via a singleton here. @tmpsantos is there an existing core object that we could set a property on?

Perhaps using a notification (& object) to request the client sets a weak delegate here?

This comment is general, and shouldn't block the PR.

Reply:
Not at the moment. But if we move at some point this abstraction to not be at the "request" level and rather at the "network manager" level it will probably solve the problem, because session becomes a property of the "network manager".